resources/js/datePicker.js

```diff
@@ -8,9 +8,9 @@ export default (initial = "") => ({
     init() {
         this.date = initial;
         let [y, m, d] = this.date.split("-");
-        this.year = y;
-        this.month = m;
-        this.day = d;
+        this.year = y ?? "";
+        this.month = m ?? "";
+        this.day = d ?? "";
```

Comment:
FOr the destructing assignment, I think you can actually set a default value. Something like:

```javascript
let [y = "", m = "", d = ""] = this.date.split("-");
```

and if so, I wonder if you could then actually do:

```javascript
[this.year = "", this.month = "", this.day =""] = this.date.split("");
```

It isn't a big deal either way though, so what is here is fine too.

Reply:
Oh I'll try that— good call, @jobara.

Reply:
You can't do the second one (destructuring to `this.year` etc). But your first example works!